components/nimbus/ios/Nimbus/NimbusApi.swift

```diff
@@ -0,0 +1,141 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+
+/**
+ * This is the main experiments API, which is exposed through the global [Nimbus] object.
+ *
+ * Application developers are encouraged to build against this API protocol, and use the `Nimbus.create` method
+ * to create the correct implementation for them.
+ *
+ * Feature developers configuring their features shoiuld use the methods in `NimbusFeatureConfiguration`. These are safe to call from any thread.
+ * Developers building UI tools for the user or QA to modify experiment enrollment will mostly use `NimbusUserConfiguration` methods.
+ * Application developers integrating `Nimbus` into their app should use the methods in `NimbusStartup`.
+ */
+public protocol NimbusApi: NimbusStartup, NimbusFeatureConfiguration, NimbusUserConfiguration {}
```

Comment:
I like being able to split up the object into named interfaces.

I do not like the name `NimbusApi`. I'd prefer to have the protocol as `Nimbus`, and the implementations be named according to their utility (`NimbusEnabled` and `NimbusDisabled`).

I haven't renamed from `NimbusApi` because it is implemented in Android with this name.